Video-Transformers/Video-Transformers/ViewController.m

```diff
@@ -0,0 +1,321 @@
+    UIImage *resizedImage = UIGraphicsGetImageFromCurrentImageContext();
+    UIGraphicsEndImageContext();
+    return resizedImage;
+}
+
+- (void)transform:(nonnull OTVideoFrame *)videoFrame {
+    
+    UIImage* image = [UIImage imageNamed:@"Vonage_Logo.png"];
+
+    uint32_t videoWidth = videoFrame.format.imageWidth;
+    uint32_t videoHeight = videoFrame.format.imageHeight;
+
+    // Calculate the desired size of the image
+    CGFloat desiredWidth = videoWidth / 8;  // Adjust this value as needed
+    CGFloat desiredHeight = image.size.height * (desiredWidth / image.size.width);
+
+    // Resize the image to the desired size
+    UIImage *resizedImage = [self resizeImage:image toSize:CGSizeMake(desiredWidth, desiredHeight)];
+
+    // Get pointer to the Y plane
+    uint8_t* yPlane = [videoFrame getPlaneBinaryData:0];
+    
+    // Create a CGContext from the Y plane
+    CGContextRef context = CGBitmapContextCreate(yPlane, videoWidth, videoHeight, 8, videoWidth, CGColorSpaceCreateDeviceGray(), kCGImageAlphaNone);
+    
+    // Location of the image (in this case right bottom corner)
+    CGFloat x = videoWidth * 4/5;
+    CGFloat y = videoHeight * 1/5;
+    
+    // Draw the resized image on top of the Y plane
+    CGRect rect = CGRectMake(x, y, desiredWidth, desiredHeight);
+    CGContextDrawImage(context, rect, resizedImage.CGImage);
+    
+    CGContextRelease(context);
+}
+
+@end
+
+customTransformer* logoTransformer;
+
+@implementation ViewController
+
+#pragma mark - View lifecycle
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    // Step 1: As the view comes into the foreground, initialize a new instance
+    // of OTSession and begin the connection process.
+    _session = [[OTSession alloc] initWithApiKey:kApiKey
+                                       sessionId:kSessionId
+                                        delegate:self];
+    [self doConnect];
+}
+
+- (BOOL)prefersStatusBarHidden
+{
+    return YES;
+}
+
+- (BOOL)shouldAutorotate {
+    return UIUserInterfaceIdiomPhone != [[UIDevice currentDevice] userInterfaceIdiom];
+}
+#pragma mark - OpenTok methods
+
+/** 
+ * Asynchronously begins the session connect process. Some time later, we will
+ * expect a delegate method to call us back with the results of this action.
+ */
+- (void)doConnect
+{
+    OTError *error = nil;
+    
+    [_session connectWithToken:kToken error:&error];
+    if (error)
+    {
+        [self showAlert:[error localizedDescription]];
+    }
+}
+
+/**
+ * Sets up an instance of OTPublisher to use with this session. OTPubilsher
+ * binds to the device camera and microphone, and will provide A/V streams
+ * to the OpenTok session.
+ */
+- (void)doPublish
+{
+    OTPublisherSettings *settings = [[OTPublisherSettings alloc] init];
+    settings.name = [UIDevice currentDevice].name;
+    _publisher = [[OTPublisher alloc] initWithDelegate:self settings:settings];
+   
+    OTError *error = nil;
+    [_session publish:_publisher error:&error];
```

Comment:
publish after we added all those transformers ?

Reply:
I don't understand what you meant. Are you asking if we need to call publish:error: after we added the transformers? If so, there is no need for that. You can add the transformers to the publisher in runtime without requiring callings publisher again

Reply:
It is more of a best practice question. Do you..

1) Init a pub, add transformers and then publish (first frame onward)
2) Init a pub, confirm that it is working with a `streamCreated` callback and then add transformers. (some delay)
3) add transformers on fly with a tap/button. (random)

The way your code is, feels like (1) .. if you do 

publish
other stuff
add transformers

with the intention of having transformers working from first frame, it may not happen because `other stuff` can be long and `streamCreated` would have started async
Not sure what the intention was..

Reply:
ah, of course. Now we have option (3). But indeed case 2 is better than 1

Reply:
Option 3 could be done in a separate button click code. Not after publish, which confuses me that it is similar to (1).
A diff click can reset.

Reply:
I introduced a button which once pressed will set or clear the transformers. There is no video transformer being set or even configured until the button is pressed.

<img width="995" alt="image" src="https://github.com/opentok/opentok-ios-sdk-samples/assets/106695857/1af2201c-fb14-48b5-b5c2-0f7ab6ea62b9">